api.go

```diff
@@ -620,41 +638,22 @@ func handleUpdateHashedKey(keyName, apiID, policyId string) (interface{}, int) {
 		sessionManager = spec.SessionManager
 	}
 
-	// This is so we bypass the hash function
-	sessStore := sessionManager.Store()
-
-	// TODO: This is pretty ugly
-	setKeyName := "apikey-" + keyName
-	rawSessionData, err := sessStore.GetRawKey(setKeyName)
-
-	if err != nil {
+	sess, ok := sessionManager.SessionDetail(keyName, true)
+	if !ok {
 		log.WithFields(logrus.Fields{
 			"prefix": "api",
 			"key":    keyName,
 			"status": "fail",
-			"err":    err,
 		}).Error("Failed to update hashed key.")
 
 		return apiError("Key not found"), 404
 	}
 
-	sess := user.SessionState{}
-	if err := json.Unmarshal([]byte(rawSessionData), &sess); err != nil {
-		log.WithFields(logrus.Fields{
-			"prefix": "api",
-			"key":    keyName,
-			"status": "fail",
-			"err":    err,
-		}).Error("Failed to update hashed key.")
-
-		return apiError("Unmarshalling failed"), 400
-	}
-
 	// Set the policy
 	sess.LastUpdated = strconv.Itoa(int(time.Now().Unix()))
 	sess.SetPolicies(policyId)
 
-	sessAsJS, err := json.Marshal(sess)
+	err := sessionManager.UpdateSession(keyName, &sess, 0)
```

Comment:
I noticed you not passing any info that key is hashed here, are you sure it will not do double hashing here?

Reply:
thats true, looks like I missed it. I was experimenting with `UpdateSession` and then decided to leave it as is because it is used a lot. I will push a fix for it